### ISSUE TITLE
get-return: false positive in get-return rule for GetInfo HTTP handler

### ISSUE THREAD

**Issue by alexandear** [mgechev/revive#1323]
The rule `get-return` incorrectly flags the HTTP handler function [`GetInfo`](https://github.com/lima-vm/lima/blob/01b435a426ba6016e0d481d83a936cdc58b2be61/pkg/hostagent/api/server/server.go#L31) as a "getter", even though it doesn't return any values in the Go sense.

Diagnostic message:
```
get-return: function 'GetInfo' seems to be a getter but it does not return any result
```

Function:
```go
func (b *Backend) GetInfo(w http.ResponseWriter, r *http.Request) {
	if r.Method != http.MethodGet {
		w.WriteHeader(http.StatusMethodNotAllowed)
		return
	}

	ctx := r.Context()
	ctx, cancel := context.WithCancel(ctx)
	defer cancel()

	info, err := b.Agent.Info(ctx)
	if err != nil {
		b.onError(w, err, http.StatusInternalServerError)
		return
	}
	m, err := json.Marshal(info)
	if err != nil {
		b.onError(w, err, http.StatusInternalServerError)
		return
	}
	w.Header().Set("Content-Type", "application/json")
	w.WriteHeader(http.StatusOK)
	_, _ = w.Write(m)
}
```

This function is an HTTP handler, and the fact that it is called `GetInfo` and uses `GET` does not mean it's a "getter" in the context of return values. The rule should be refined to avoid this false positive.

**Comment by alexandear** [comment]
`GetInfo` is not a getter because the prefix `Get` means `GET` HTTP method.

**Comment by chavacava** [comment]
Yes, rules might produce false positives. If these false positives are too frequent then it's a problem and the rule implementation must address it, if not, just let the rule as is and add a `//revive:disable...`

**Comment by ccoVeille** [comment]
For record, I never applied this rule at work because of the exact issue @alexandear reported with [http.Handler](https://pkg.go.dev/net/http#Handler), or [http.HandlerFunc](https://pkg.go.dev/net/http#HandlerFunc)

Sometimes, there are also in-house wrapper that adds other parameters like `func(w http.ResponseWriter, r *http.Request, authUser Whatever)` 

There might also be false positive with gin, chi, and others frameworks, but here it could be about adding a configuration option

**Comment by chavacava** [comment]
[How frequent is this pattern in github repos?](https://github.com/search?q=%2Ffunc%5Cs*%28%5C%28.*%5C%29%29%3F%5Cs*Get.*%5C%28w+http%5C.ResponseWriter%2C+r+%5C*http%5C.Request%5C%29%2F+path%3A*.go+language%3AGo&type=code&ref=advsearch)

**Comment by ccoVeille** [comment]
> For record, I never applied this rule at work because of the exact issue @alexandear reported with [http.Handler](https://pkg.go.dev/net/http#Handler), or [http.HandlerFunc](https://pkg.go.dev/net/http#HandlerFunc)
> 
> Sometimes, there are also in-house wrapper that adds other parameters like `func(w http.ResponseWriter, r *http.Request, authUser Whatever)` 
> 
> There might also be false positive with gin, chi, and others frameworks, but here it could be about adding a configuration option

Some stats about this pattern was provided [here](https://github.com/mgechev/revive/pull/1325#discussion_r2061230388) when reviewing

- https://github.com/mgechev/revive/pull/1325

**Comment by ccoVeille** [comment]
Here is what have been implemented.

> Any getter method/function that returns nothing and first two parameters are "http.ResponseWriter, *http.Request" in that order is ignored
> Source : https://github.com/mgechev/revive/pull/1325#issuecomment-2833386110
